init.sls

```diff
@@ -106,20 +106,48 @@ chown_pgdata:
 # Deploy users. Sort them, so order is not changing between salt runs
 {% for config in pillar['postgresql']['users']|default({}) %}
 createuser-{{ loop.index }}:
-  cmd.run:
-    - name: psql -t -c "CREATE ROLE {{ config['username'] }} {{ config['options']|default('NOSUPERUSER NOCREATEDB NOCREATEROLE INHERIT LOGIN') }};"
-    - unless: psql -t -c "SELECT 1 FROM pg_roles WHERE rolname='{{ config['username'] }}'" |grep -q 1
-    - runas: postgres
+  postgres_user.present:
+    - name: {{ config['username'] }}
+    - encrypted: True
+    {% if config['login']|default(false) %}
+    - login: True
+    {% endif %}
+    {% if config['password'] is defined %}
+    - password: {{ config['password'] }}
+    {% endif %}
+    {% if config['createdb']|default(false) %}
+    - createdb: True
+    {% endif %}
+    {% if config['inherit']|default(false) %}
+    - inherit: True
+    {% endif %}
+    {% if config['replication']|default(false) %}
+    - replication: True
+    {% endif %}
```

Comment:
I think we should also support the other [options](https://docs.saltstack.com/en/latest/ref/states/all/salt.states.postgres_user.html): superuser, createroles, createdb